File: Leetcode/Typescript/2914. Minimum Number of Changes to Make Binary String Beautiful/solution.ts
// https://leetcode.com/problems/minimum-number-of-changes-to-make-binary-string-beautiful/description

function minChanges(s: string): number {
  let changeCount = 0;
  // Divide string minimum even length and check it needs change
  for (let idx = 0; idx < s.length; idx += 2) {
    changeCount += s[idx] !== s[idx + 1] ? 1 : 0;
  }
  return changeCount;
}

File: Leetcode/Python/2914. Minimum Number of Changes to Make Binary String Beautiful/solution.py
# https://leetcode.com/problems/minimum-number-of-changes-to-make-binary-string-beautiful/description

class Solution:  # First solution
    def minChanges(self, s: str) -> int:
        change_count = 0
        # Divide string minimum even length and count min change
        for idx in range(0, len(s), 2):
            zero = s[idx: idx + 2].count("0")
            one = 2 - zero
            change_count += min(zero, one)

        return change_count


class Solution:  # Second solution
    def minChanges(self, s: str) -> int:
        change_count = 0
        # Divide string minimum even length and check it needs change
        for idx in range(0, len(s), 2):
            change_count += s[idx] != s[idx + 1]

        return change_count
